actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -306,7 +306,7 @@ def label_tag(name = nil, content_or_options = nil, options = nil, &block)
       #   # => <input type="hidden" name="collected_input" id="collected_input"
       #        value="" onchange="alert(&#39;Input collected!&#39;)" autocomplete="off" />
       def hidden_field_tag(name, value = nil, options = {})
-        text_field_tag(name, value, options.merge(type: :hidden, autocomplete: "off"))
+        text_field_tag(name, value, options.merge(type: :hidden).reverse_merge(autocomplete: "off"))
```

Comment:
Since the preceding `options.merge` call will create a new `Hash` instance, would it make sense to invoke the `!` variant?

```suggestion
        text_field_tag(name, value, options.merge(type: :hidden).reverse_merge!(autocomplete: "off"))
```

Reply:
Could the [Hash#with_defaults!](https://edgeapi.rubyonrails.org/classes/Hash.html#method-i-with_defaults-21) alias work instead?

```suggestion
        text_field_tag(name, value, options.merge(type: :hidden).with_defaults!(autocomplete: "off"))
```

Reply:
Interesting. I didn't know about that alias. It does read better :)

Is there a reason not to use `merge!` also?